src/app/components/course-island/course-island.component.ts

```diff
@@ -1,14 +1,29 @@
-import {Component, Input} from '@angular/core'
-import {Course, STATUS} from "@app/_models"
+import {Component, Input, OnInit} from '@angular/core'
+import {Course, STATUS, User} from "@app/_models"
+import {AuthenticationService} from "@app/_services/api/authentication"
+
 
 @Component({
     selector: 'app-course-island',
     templateUrl: './course-island.component.html',
     styleUrls: ['./course-island.component.scss']
 })
-export class CourseIslandComponent {
+export class CourseIslandComponent implements OnInit{
     @Input() course: Course
     @Input() hasViewPermission: boolean
     @Input() skeleton = false
+    user: User
     readonly STATUS = STATUS
+    currentDate = new Date()
```

Comment:
currentDate should be initialized in the `ngOninit`.

Reply:
all done

Reply:
understood